docs/usermanual.md

```diff
@@ -315,84 +315,72 @@ All the methods **must** call the callback before exiting (with or without error
 IoT Agent Node.js library to interact process incoming requests.
 
 ## Development documentation
+
 ### Project build
-The project is managed using Grunt Task Runner.
+The project is managed using npm.
 
 For a list of available task, type
 ```bash
-grunt --help
+npm run
 ```
 
 The following sections show the available options in detail.
 
-### Coding guidelines
-jshint, gjslint
+### Testing
+[Mocha](http://visionmedia.github.io/mocha/) Test Runner + [Should.js](https://shouldjs.github.io/) Assertion Library.
+
+The test environment is preconfigured to run BDD testing style.
+
+Module mocking during testing can be done with [proxyquire](https://github.com/thlorenz/proxyquire)
+
+To run tests, type
 
-Uses provided .jshintrc and .gjslintrc flag files. The latter requires Python and its use can be disabled
-while creating the project skeleton with grunt-init.
-To check source code style, type
 ```bash
-grunt lint
+npm test
 ```
 
-Checkstyle reports can be used together with Jenkins to monitor project quality metrics by means of Checkstyle
-and Violations plugins.
-To generate Checkstyle and JSLint reports under `report/lint/`, type
+### Coding guidelines
+jshint
+
+Uses provided .jshintrc flag file.
+To check source code style, type
+
 ```bash
-grunt lint-report
+npm run lint
 ```
 
-
 ### Continuous testing
 
 Support for continuous testing by modifying a src file or a test.
 For continuous testing, type
+
 ```bash
-grunt watch
+npm run test:watch
 ```
 
+If you want to continuously check also source code style, use instead:
 
-### Source Code documentation
-dox-foundation
-
-Generates HTML documentation under `site/doc/`. It can be used together with jenkins by means of DocLinks plugin.
-For compiling source code documentation, type
 ```bash
-grunt doc
+npm run watch
 ```
 
-
 ### Code Coverage
 Istanbul
 
 Analizes the code coverage of your tests.
 
 To generate an HTML coverage report under `site/coverage/` and to print out a summary, type
-```bash
-# Use git-bash on Windows
-grunt coverage
-```
 
-To generate a Cobertura report in `report/coverage/cobertura-coverage.xml` that can be used together with Jenkins to
-monitor project quality metrics by means of Cobertura plugin, type
 ```bash
 # Use git-bash on Windows
-grunt coverage-report
-```
-
-
-### Code complexity
-Plato
-
-Analizes code complexity using Plato and stores the report under `site/report/`. It can be used together with jenkins
-by means of DocLinks plugin.
-For complexity report, type
-```bash
-grunt complexity
+npm run test:coverage
 ```
 
-### PLC
+### Clean
 
+<<<<<<< HEAD
```

Comment:
Ups... seems a conflict resolution lefover ;)

Reply:
Too many PR for a single day! :P Fixed with 0cdfdd2c211574bc2cc3ffa8af44be3ad22ae3c5